src/queries/main.ts

```diff
@@ -0,0 +1,55 @@
+import User from '@/shapes/model/session/User';
+import { useInfiniteQuery, useQuery } from '@tanstack/react-query';
+import axios from 'axios';
+
+const fetchFeeds = async (userId: number, date: Date) => {
+  const dateString = date.toJSON().slice(0, 10);
+
+  const response = await axios.get(`/api/users/${userId}/feeds`, {
+    params: { date: dateString },
+    metadata: {
+      gaCategory: 'Feed',
+      gaVariable: 'GET / List',
+    },
+  });
+  return response.data;
+};
+
+const fetchNotices = async () => {
+  const now = new Date();
+  const response = await axios.get('/api/notices', {
+    params: {
+      time: now.toJSON(),
+      order: ['start_time', 'id'],
+    },
+    metadata: {
+      gaCategory: 'Notice',
+      gaVariable: 'GET / List',
+    },
+  });
+  return response.data;
+};
+
+export const useFeed = (user: User) => {
+  const userId = user?.id;
```

Comment:
user를 param으로 넣어주는 대신 useSessionInfo 사용하면 어떨까요?

```suggestion
export const useFeed = (user: User) => {
  const { data: user } = useSessionInfo();
```

Reply:
@withSang 
-  이부분은 그럼 user가 없을 때 처리 추가하기 
(기존과 동일하게 get session/info 의 status code 가 401 이면 user 를 null 로 설정하는 로직을 useUserSession의 query Fn 에 넣기)

-  새로 작성된 컴포넌트에서는 redux store 에서가 아닌 useSessionInfo 를 통해서 user Session 가져오기 로 변경하면 어떨까요?